src/RendezvousTracker.ts

```diff
@@ -0,0 +1,268 @@
+                // Check for both the .brs and .xml versions of the file starting with .brs
+                fileNameAsBrs = this.convertDebuggerPathToClient(fileNameAsBrs);
+                if (fileNameAsBrs) {
+                    fileName = fileNameAsBrs;
+                } else {
+                    fileNameAsXml = this.convertDebuggerPathToClient(fileNameAsXml);
+                    if (fileNameAsXml) {
+                        fileName = fileNameAsXml;
+                    }
+                }
+            }
+
+            this.clientPathsMap[fileName] = {
+                clientPath: this.convertDebuggerPathToClient(fileName),
+                clientLines: {
+                    [lineNumber]: this.convertDebuggerLineToClientLine(fileName, parseInt(lineNumber))
+                }
+            };
+        } else if (!this.clientPathsMap[fileName].clientLines[lineNumber]) {
+            // Add new client line to clint path map
+            this.clientPathsMap[fileName].clientLines[lineNumber] = this.convertDebuggerLineToClientLine(fileName, parseInt(lineNumber));
+        }
+
+        return fileName;
+    }
+
+    /**
+     * Helper function to assist in the creation of a RendezvousLineInfo
+     * @param fileName processed file name
+     * @param lineNumber occurrence line number
+     * @param duration how long the rendezvous took to complete, if not supplied it is assumed to be zero
+     */
+    private createLineObject(fileName: string, lineNumber: number, duration?: string): RendezvousLineInfo {
+        return {
+            clientLineNumber: lineNumber,
+            clientPath: this.clientPathsMap[fileName].clientPath,
+            hitCount: 1,
+            totalTime: this.getTime(duration),
+            type: 'lineInfo'
+        };
+    }
+
+    /**
+     * Helper function to convert the duration to a float or return 0.00
+     * @param duration how long the rendezvous took to complete, if not supplied it is assumed to be zero
+     */
+    private getTime(duration?: string): number {
+        return duration ? parseFloat(duration) : 0.000;
+    }
+
+    /**
+     * Replaces the last occurrence of a string in a string
+     * @param base String to search within
+     * @param replace String to replace
+     * @param replacement What to replace the final occurrence with
+     */
+    private replaceLastStringOccurrence(base: string, replace: string, replacement: string): string {
+        return (base.includes(replace)) ? base.substr(0, base.lastIndexOf(replace)) + replacement : base;
+    }
+}
+
+/**
+ * Used to see if a field on a object in RendezvousHistory object is a reserved field
+ * @param fieldName name of the field to check
+ */
+export function isRendezvousDetailsField(fieldName: string): boolean {
+    return (fieldName === 'type' || fieldName === 'hitCount' || fieldName === 'totalTime' || fieldName === 'zeroCostHitCount');
+}
+
+export interface RendezvousHistory {
```

Comment:
Delete this interface and instead, just use `{[key: string]: RendezvousFileInfo}` where it's defined above.

Reply:
This interface exists as it is used in the RendezvousViewProvider.

Reply:
Updated to make this interface more useful down the line.